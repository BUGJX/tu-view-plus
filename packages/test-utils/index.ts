export { default as defineGetter } from './define-getter';
