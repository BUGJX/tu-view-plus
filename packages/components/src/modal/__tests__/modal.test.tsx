import { nextTick } from 'vue'
import { mount } from '@vue/test-utils'
import { describe, expect, test, vi } from 'vitest'
import Modal from '../index'
import ModalComponent from '../src/modal.vue'

const AXIOM = 'Tu view is good'

describe('Modal', () => {
  test('should render modal and emit ok/cancel event', async () => {
    const wrapper = mount(ModalComponent, {
      props: {
        defaultVisible: true,
        renderToBody: false,
      },
      slots: {
        default: `<div>${AXIOM}</div>`,
        title: 'Title',
      },
    });

    await nextTick();
    expect(wrapper.html()).toMatchSnapshot();
    const buttons = wrapper.findAll('button');
    await buttons[0].trigger('click');
    expect(wrapper.emitted('cancel')).toHaveLength(1);
    await buttons[1].trigger('click');
    expect(wrapper.emitted('ok')).toHaveLength(1);
  });

  test('should show modal and call cb', async () => {
    const onOk = vi.fn()
    const onCancel = vi.fn()

    const wrapper = mount({
      template: `<button @click="handleClick">Click</button>`,
      methods: {
        handleClick() {
          Modal.open({
            title: 'title',
            content: 'content',
            onOk,
            onCancel,
          });
        },
      },
    });

    await wrapper.find('button').trigger('click');
    const buttons = document.querySelectorAll('button');
    await buttons[0]?.click();
    expect(onCancel).toBeCalled();
    await buttons[1]?.click();
    expect(onOk).toBeCalled();
  });
})
