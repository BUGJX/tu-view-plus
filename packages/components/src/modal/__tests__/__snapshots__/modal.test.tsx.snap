// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Modal > should render modal and emit ok/cancel event 1`] = `
"<!--teleport start-->
<div class=\\"tu-modal-container\\" style=\\"z-index: 1001;\\">
  <div class=\\"tu-modal-container__mask fade-modal-enter-from fade-modal-enter-active\\"></div>
  <div class=\\"tu-modal-wrapper is-align-center\\">
    <div class=\\"tu-modal tu-modal--medium zoom-modal-enter-from zoom-modal-enter-active\\">
      <!-- modal header -->
      <div class=\\"tu-modal__header\\">
        <div class=\\"tu-modal__title is-align-start\\">
          <!--v-if-->Title
        </div>
        <div tabindex=\\"-1\\" role=\\"button\\" aria-label=\\"Close\\" class=\\"tu-modal__icon-close\\"><i class=\\"tu-icon\\">
            <!--v-if--><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 1024 1024\\">
              <path fill=\\"currentColor\\" d=\\"M764.288 214.592 512 466.88 259.712 214.592a31.936 31.936 0 0 0-45.12 45.12L466.752 512 214.528 764.224a31.936 31.936 0 1 0 45.12 45.184L512 557.184l252.288 252.288a31.936 31.936 0 0 0 45.12-45.12L557.12 512.064l252.288-252.352a31.936 31.936 0 1 0-45.12-45.184z\\"></path>
            </svg>
            <!--v-if-->
          </i></div>
      </div><!-- modal body -->
      <div class=\\"tu-modal__body\\">
        <div>Tu view is good</div>
      </div><!-- modal footer -->
      <div class=\\"tu-modal__footer\\"><button ariadisabled=\\"false\\" type=\\"button\\" class=\\"tu-button tu-button--medium tu-button--default\\">
          <!--v-if-->
          <!--v-if--><span>Cancel</span>
        </button><button ariadisabled=\\"false\\" type=\\"button\\" class=\\"tu-button tu-button--medium tu-button--primary\\">
          <!--v-if-->
          <!--v-if--><span>OK</span>
        </button></div>
    </div>
  </div>
</div>
<!--teleport end-->"
`;
